app/customer.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass
+import math
+
+from app.car import Car
+from app.shop import Shop
+
+
+@dataclass
+class Customer:
+    name: str
+    products_cart: dict
+    location: list[int]
+    money: int | float
+    car: Car
+
+    def have_money(self) -> str:
+        return f"{self.name} has {self.money} dollars"
+
+    def trip_cost(self, shop: Shop, fuel_price: float) -> int | float:
+        distance = math.sqrt(
+            (self.location[0] - shop.location[0]) ** 2
+            + (self.location[1] - shop.location[1]) ** 2)
+        fuel = distance / 100 * self.car.fuel_consumption
+        return round(fuel * fuel_price * 2, 2)
+
+    def product_cost(self, shop: Shop) -> int | float:
+        if not shop.products:
+            return 0
+        product_price = 0
+        for key, value in self.products_cart.items():
+            if key in shop.products:
+                product_price += value * shop.products[key]
+            else:
+                raise KeyError("There is no required product")
+        return product_price
+
+    def travel_calculation(self, shop: Shop, fuel_price: float) -> int | float:

```

Comment:
The method `travel_calculation` calculates the total cost of a trip and products. Ensure that the method name and its purpose are clearly documented, as it might not be immediately clear what 'travel_calculation' entails.